src/vsr/replica.zig

```diff
@@ -208,18 +208,27 @@ pub fn ReplicaType(
         /// we require and assert in our protocol implementation.
         loopback_queue: ?*Message = null,
 
-        /// Unique start_view_change messages for the same view from OTHER replicas (excluding ourself).
-        start_view_change_from_other_replicas: QuorumCounter = quorum_counter_null,
+        /// The last timestamp received on a commit heartbeat.
+        /// Used to discard delayed heartbeat messages.
+        /// (status=normal backup)
+        heartbeat_timestamp: u64 = 0,
```

Comment:
Could you say in the comment which clock is that? I can see at least two choices: 

* replica's local time
* our cluster time

(and I don't yet fully understand cluster time)

Reply:
:+1: I'll add a comment. Fyi it is the primary's monotonic clock — all we are about here is the ability to discard old/duplicate messages.

Reply:
Yeah, I think a `(view, primary.monotonic)` is enough for discarding. We need view though, to avoid comparing clocks of different primaries.

Reply:
Yeah, we ignore any `command=commit` messages from a different view.